### PR TITLE
chore: remove dependency on unmaintained 'errors' package (#490)

### LINKED ISSUE #490: github.com/pkg/errors is no longer maintained
pkg/errors is no longer maintained. See pkg/errors#245

We can replaces all calls to errors.Wrap/f with fmt.Errorf and removes the unneeded dependency on
github.com/pkg/errors since we don't support go 1.12 or lower. See microsoft/go-winio#227 for an example.

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Fixes #490

### CODE REVIEW

**Review by blkperl — APPROVED** [review]

**Review by ishitasequeira — APPROVED** [review]
LGTM

File: test/e2e/fixture/applicationsets/utils/fixture.go [hunk 1]
```diff
@@ -114,7 +113,7 @@ func EnsureCleanState(t *testing.T) {
 			// Fail
 			msg := fmt.Sprintf("Waiting for list of ApplicationSets to be size zero: %d", len(list.Items))
 			// Intentionally not making this an Errorf, so it can be printf-ed for debugging purposes.
-			return errors.New(msg)
+			return fmt.Errorf(msg)
```
Comment (wtam2018) [inline test/e2e/fixture/applicationsets/utils/fixture.go:116]: Should we clean this (and other ones in this file) up to combine them into a single fmt.Errorf() call?  If we debug with printf, we would've to edit it anyway.

Comment (crenshaw-dev) [reply test/e2e/fixture/applicationsets/utils/fixture.go:116]: Agreed, will update!

**Review by wtam2018 — APPROVED** [review]
LGTM
thanks @crenshaw-dev !

**Pull request merged**